download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadViewModel.kt

```diff
@@ -157,9 +157,18 @@ class DownloadViewModel(
 
   private fun install(resolver: ConstraintsResolver) {
     viewModelScope.launch {
-      val installPackageInfo = installPackageInfoMapper.map(app)
-      resolver(appInstaller.canInstall(installPackageInfo)) {
-        install(installPackageInfo, it)
+      val installPackageInfo: InstallPackageInfo? = try {
+        installPackageInfoMapper.map(app)
+      } catch (e: Exception) {
+        viewModelState.update {
+          DownloadUiState.Error(retryWith = ::install)
+        }
+        null
+      }
+      installPackageInfo?.let {
```

Comment:
NIT: You can attach this directly to the try-catch and avoid using a temporary `installPackageInfo` value

Reply:
Fixed